api/src/main/java/org/openmrs/module/m2sysbiometrics/service/impl/NationalSynchronizationFailureServiceImpl.java

```diff
@@ -18,4 +20,26 @@
     public NationalSynchronizationFailure save(NationalSynchronizationFailure nationalSynchronizationFailure) {
         return dao.save(nationalSynchronizationFailure);
     }
+
+    @Override
+    public NationalSynchronizationFailure saveOrReplaceRegistrationFailure(
+            NationalSynchronizationFailure nationalSynchronizationFailure) {
+        if (nationalSynchronizationFailure.getBiometricXml() != null) {
+            for (NationalSynchronizationFailure failureFromDB : dao.findAllRegistrationFailuresByBiometricXml(
```

Comment:
this is a poor solution, since the bimetric xml will warry between scans. What you would have to do here is a biometric match, which would be rewriting the m2sys solution.

I don't really foresee the same fingerprint getting registered multiple times - is there a use case for this check?

Reply:
The same fingerprint is getting registered multiple times if the retry of national registration fails.
To be more precise, a use case is something like:
1. Trying to register nationally, but there's no connection with the national server.
2. Scheduled task retry this enrollment, but it failed again.

Reply:
In my opinion it should be implemented in this way:
1. Client tries to register to server 
2. Server responds with failure
3. Registration-retry task is added to FIDO queue
4. Scheduler executes the task body (the task is not deleted from queue)
5. Registration is conducted again
- if failure the task is not deleted
- if success the task is deleted

Reply:
Yes, I think that's the best way to handle this case.